services/rpcfilters/transaction_sent_to_upstream_event_test.go

```diff
@@ -0,0 +1,81 @@
+package rpcfilters
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+)
+
+var transactionHashes = []common.Hash{common.HexToHash("0xAA"), common.HexToHash("0xBB"), common.HexToHash("0xCC")}
+
+func TestTransactionSentToUpstreamEventSubscribe(t *testing.T) {
+	event := newTransactionSentToUpstreamEvent()
+	assert.NoError(t, event.Start())
+	defer event.Stop()
+
+	_, channel := event.Subscribe()
+
+	done := make(chan struct{})
+	go func() {
+		for i, expectedHash := range transactionHashes {
+			select {
+			case receivedHash := <-channel:
+				assert.Equal(t, expectedHash, receivedHash)
+			case <-time.After(1 * time.Second):
+				assert.Fail(t, "timeout")
+			}
+			if i == len(transactionHashes)-1 {
+				close(done)
+			}
+		}
+	}()
+
+	for _, hashToTrigger := range transactionHashes {
+		// sleep in order to ensure those hashes come in order
+		time.Sleep(10 * time.Millisecond)
+		event.Trigger(hashToTrigger)
+	}
+	<-done
+}
+
+func TestTransactionSentToUpstreamEventMultipleSubscribe(t *testing.T) {
```

Comment:
I think you can keep only this test case and get rid of `TestTransactionSentToUpstreamEventSubscribe` which is the same but for a case with a single subscriber.